Framework/Core/src/HistogramRegistry.cxx

```diff
@@ -95,24 +96,25 @@ void HistogramRegistry::validateHistName(const char* name, const uint32_t hash)
    */
 }
 
-void HistogramRegistry::add(const HistogramSpec& histSpec)
+HistPtr HistogramRegistry::add(const HistogramSpec& histSpec)
 {
-  insert(histSpec);
+  return insert(histSpec);
 }
 
-void HistogramRegistry::add(char const* const name, char const* const title, const HistogramConfigSpec& histConfigSpec, bool callSumw2)
+HistPtr HistogramRegistry::add(char const* const name, char const* const title, const HistogramConfigSpec& histConfigSpec, bool callSumw2)
 {
-  insert({name, title, histConfigSpec, callSumw2});
+  return insert({name, title, histConfigSpec, callSumw2});
 }
 
-void HistogramRegistry::add(char const* const name, char const* const title, HistType histType, std::vector<AxisSpec> axes, bool callSumw2)
+HistPtr HistogramRegistry::add(char const* const name, char const* const title, HistType histType, std::vector<AxisSpec> axes, bool callSumw2)
 {
-  insert({name, title, {histType, axes}, callSumw2});
+  return insert({name, title, {histType, axes}, callSumw2});
 }
 
 // store a copy of an existing histogram (or group of histograms) under a different name
-void HistogramRegistry::addClone(const std::string& source, const std::string& target)
+HistPtr HistogramRegistry::addClone(const std::string& source, const std::string& target)
```

Comment:
This function is able also to clone whole directory structures. 
The pointer that will be returned here in this case would therefore be unpredictable (it will be the last histogram in the registry that matches the request).
My suggestion would be to keep this as a void function for now and later we can split it into a version for cloning single histograms that returns the corresponding pointer and a version that we call `addGroupClone` or similar which will then be a `void` function.

Reply:
Hi @mpuccio and @mario-krueger,
Hi think that Mario's suggestion is reasonable. Can you solve this so we can merge?

Reply:
we can also merge it as it is and I will then commit the follow-up changes

Reply:
Yeah, sorry I got back from vacation today! I am fixing it right now :)